backend/Application/Observability/ApplicationMetrics.cs

```diff
@@ -0,0 +1,126 @@
+using System.Diagnostics;
+using System.Text;
+using HotChocolate.Execution;
+using Microsoft.Extensions.ObjectPool;
+using Prometheus;
+
+namespace Application.Observability;
+
+internal static class ApplicationMetrics
+{
+    private static readonly Histogram GraphQlRequestDuration = Metrics.CreateHistogram(
+        "graphql_request_duration_seconds",
+        "Duration of GraphQl in seconds",
+        new HistogramConfiguration
+        {
+            LabelNames = new[] { "operation", "exception" }
+        });
+    
+    internal class GraphQlDurationMetric : IDisposable
```

Comment:
This entire class seems very low level. Have you looked for a library that might do the same and can hide away all of this?

Reply:
Also can't find any test.

Reply:
I could only find this
https://chillicream.com/docs/hotchocolate/v13/server/instrumentation

Which mostly refer to tracing. Since we are not using either Jaeger or Zipkins this isn’t for us.

OpenTelemetry does have some experimental features exporting to Prometheus, but this is something I haven’t used
https://opentelemetry.io/docs/instrumentation/net/exporters/#prometheus-experimental

Looking at Hot Chocolate (our GraphQL framework) using telemetry extensions would give us metrics. Their extensions are defined here
https://github.com/ChilliCream/graphql-platform/blob/c97b49b5b7c624ee7acecc0111f8fbfdf0c1b279/src/HotChocolate/Diagnostics/src/Diagnostics/Extensions/DiagnosticsRequestExecutorBuilderExtensions.cs#L28C24-L28C24

Which only add diagnostics listeners
https://github.com/ChilliCream/graphql-platform/blob/c97b49b5b7c624ee7acecc0111f8fbfdf0c1b279/src/HotChocolate/Diagnostics/src/Diagnostics/Extensions/DiagnosticsRequestExecutorBuilderExtensions.cs#L69

Looking at these they enrich activities, but doesn’t report any metrics. Ex.
- https://github.com/ChilliCream/graphql-platform/blob/c97b49b5b7c624ee7acecc0111f8fbfdf0c1b279/src/HotChocolate/Diagnostics/src/Diagnostics/Listeners/ActivityExecutionDiagnosticListener.cs#L60
- https://github.com/ChilliCream/graphql-platform/blob/main/src/HotChocolate/Diagnostics/src/Diagnostics/Scopes/ExecuteRequestScope.cs
- https://github.com/ChilliCream/graphql-platform/blob/c97b49b5b7c624ee7acecc0111f8fbfdf0c1b279/src/HotChocolate/Diagnostics/src/Diagnostics/Scopes/RequestScopeBase.cs#L35C12-L35C12
- https://github.com/ChilliCream/graphql-platform/blob/c97b49b5b7c624ee7acecc0111f8fbfdf0c1b279/src/HotChocolate/Diagnostics/src/Diagnostics/ActivityEnricher.cs#L345

Having it testable will break the encapsulation which I really want to avoid. I have, and will do some manual testing validating metrics.